src/DoubleTheDonation/FormExtension/Actions/FieldScope/PaymentMeta.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace GiveDoubleTheDonation\DoubleTheDonation\FormExtension\Actions\FieldScope;
+
+use Closure;
+use Give\Donations\Models\Donation;
+use GiveDoubleTheDonation\DoubleTheDonation\FormExtension\Field as DoubleTheDonationField;
+
+/**
+ * Save payment meta - field scope action
+ *
+ * @unreleased
+ */
+class PaymentMeta
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke(): Closure
+    {
+        // Scope callback
+        return function (DoubleTheDonationField $field, $company, Donation $donation) {
```

Comment:
Woah this is an interesting approach. I was imaging the invoke method would receive the callback params - however I suppose if this is being passed directly to the scope than we would need the return callback.

Reply:
I want to extend the Fields API to directly support passing an action class. Currently, you can only pass closures and strings. That's okay if you have really simple stuff to handle like updating one meta field, but anything more complex simply does not look good on the field definition.